### PR TITLE
Update BoringSSL to 5298ef99bf2b2d77600b3bb74dd572027bf495be

### PR DESCRIPTION
**Pull request by Lukasa**
Note that the vendor script changed: BoringSSL added a new top-level
header that had to be added to the umbrella header.

### CODE REVIEW

**Comment by weissi** [comment]
weirdo error

```
18:04:39 [354/378] Compiling NIOSSL IdentityVerification.swift
18:04:39 <unknown>:0: warning: missing submodule 'CNIOBoringSSLShims'
18:04:39 /tmp/.check-api_BnXsXX/repo/Sources/NIOSSL/SSLCertificate.swift:17:29: error: no such module 'CNIOBoringSSLShims'
18:04:39 @_implementationOnly import CNIOBoringSSLShims
18:04:39                             ^
18:04:39 [355/378] Compiling NIOSSL LinuxCABundle.swift
18:04:39 <unknown>:0: warning: missing submodule 'CNIOBoringSSLShims'
18:04:39 /tmp/.check-api_BnXsXX/repo/Sources/NIOSSL/SSLCertificate.swift:17:29: error: no such module 'CNIOBoringSSLShims'
18:04:39 @_implementationOnly import CNIOBoringSSLShims
18:04:39                             ^
```

**Comment by weissi** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

File: Sources/CNIOBoringSSL/crypto/fipsmodule/bn-586.linux.x86.S [hunk 1]
```diff
@@ -995,553 +995,6 @@ bn_sub_words:
...
-.L052pw_tail_nc6:
-.L054pw_nc_end:
-	movl	$0,%eax
-.L032pw_end:
-	popl	%edi
-	popl	%esi
-	popl	%ebx
-	popl	%ebp
-	ret
-.size	bn_sub_part_words,.-.L_bn_sub_part_words_begin
```
Comment (Davidde94) [inline Sources/CNIOBoringSSL/crypto/fipsmodule/bn-586.linux.x86.S]: As I'm still relatively new, why do we check-in the assembly?

Comment (Lukasa) [reply Sources/CNIOBoringSSL/crypto/fipsmodule/bn-586.linux.x86.S]: The assembly is generated by a tool called `perlasm`, which takes the `.pl` files and processes them into the appropriate assembly for the target. Swift PM cannot run scripts at install time, so we cannot do this work on the client machine. As a result, we have to do it ahead-of-time and consider them part of the source artefact.

Comment (Davidde94) [reply Sources/CNIOBoringSSL/crypto/fipsmodule/bn-586.linux.x86.S]: Makes sense, thanks :)

**Pull request merged**